apidoc/Titanium/Geolocation/Geolocation.yml

```diff
@@ -282,6 +282,63 @@ methods:
     platforms: [iphone, ipad, android]
     since: "5.1.0"
 
+  - name: requestTemporaryFullAccuracyAuthorization
```

Comment:
Do we have to create a new method? From a cross-platform standpoint, it might be nice to re-use our existing [requestLocationPermissions()](https://docs.appcelerator.com/platform/latest/#!/api/Titanium.Geolocation-method-requestLocationPermissions) and [hasLocationPermissions()](https://docs.appcelerator.com/platform/latest/#!/api/Titanium.Geolocation-method-hasLocationPermissions) methods and add the new "purpose" string key as a parameter... or even better add support for a settings dictionary to these methods with a new "purpose" field.

On Android, location permission can be granted temporarily by the end-user too. We can still leverage the `hasLocationPermission()` method to check if it's still granted. And the `Ti.Location.accuracy` property defines the accuracy of the permission we need to look for. On Android, the `ACCESS_FINE_LOCATION` sounds like the equivalent of the "accurate" permission on iOS (which means read from the GPS).
https://developer.android.com/preview/privacy/permissions#one-time

So, can we do something like this?
```javascript
const permissionSettings = {
	authorization: Ti.Geolocation.AUTHORIZATION_ALWAYS,
	purpose: 'PurposeKey1',
};
if (Ti.Geolocation.hasLocationPermissions(permissionSettings)) {
	fetchLocationInfo();
} else {
	Ti.Geolocation.requestLocationPermissions(permissionSettings, (e) => {
		if (e.success) {
			fetchLocationInfo();
		}
	});
}
```

**Question:**
On iOS 14, is the "purpose" key required? Are people going to have location accessing problem if they don't update their location permission handling code?

Reply:
@jquick-axway This new API is for requesting for full accuracy if user requires it in app. And it get expires. It is not mandatory. e.g your app do not need  full accurate location, you should simply need location access which will not expire. let's say at some screen you need full accuracy to show use'r location, you have to go for it. See [here](https://developer.apple.com/documentation/corelocation/cllocationmanager/3600217-requesttemporaryfullaccuracyauth?language=objc) for more details

'purpose' key is only required if one is going to access location with full accuracy. It will not break existing implementation.

Reply:
So my only beef with this is if the app developer sets `Ti.Location.accuracy` to `ACCURACY_HIGH`, then I think `hasLocationPermissions()` should return `false` if it has high accuracy has expired, because that's what the app desires. That's why I'm question the necessity of these new APIs... that and I think we're making this more complicated for app developers than it needs to be. A portable API that works between all iOS versions and Android would be the ideal.

Reply:
If user has granted location permission `hasLocationPermissions()` will be `true`. ACURRACY_HIGH (kCLLocationAccuracyBest) is desirable not guaranteed.
In iOS 14+ when app request for location permission there comes a popup (see attached Image1) which have a button mentioning 'Precise: On'. If user sets it off, app will not get location data with full accuracy. If at some point it is required to get 'full accuracy', with this new API app can ask for permission to get full accuracy in location data with a valid reason (see image2 ). And it will be temporary.
Image1- ![IMG_1632](https://user-images.githubusercontent.com/22900907/91914367-bd03fe00-ec6c-11ea-965e-18da6cf9fa0b.PNG)

image2 - 
![IMG_1634](https://user-images.githubusercontent.com/22900907/91914437-e7ee5200-ec6c-11ea-9c62-425479221db0.PNG)